### PR TITLE
Dataservice

### PR DESCRIPTION
**Pull request by kruplm**

### CODE REVIEW

**Review by nick-polyak-ms — COMMENTED** [review]
I wonder if it is possible to move the generic code from hackday1 into the generic projects - e.g. we should probably move the message router under tryouts/core or plugins. I'll be happy to help with the integration. 
Some of the test code can also go into the plugins and some - into the prototype. Let us have a  hacking session (perhaps next week) to integrate the hackday code into the code projects under Tryouts.

**Review by psmulovics — APPROVED** [review]

File: Tryouts/Prototypes/hackday1/ComposeUI.Example.DataService/Publisher.cs [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using ComposeUI.Messaging.Client;
+using ComposeUI.Messaging.Client.Transport.WebSocket;
+using Microsoft.Extensions.Logging;
+
+namespace ComposeUI.Example.DataService
+{
+    internal class SubscriptionsMonitor : IObserver<RouterMessage>
```
Comment (ZKRobi) [inline Tryouts/Prototypes/hackday1/ComposeUI.Example.DataService/Publisher.cs:27]: In general classes should be in their own files

**Review by ZKRobi — DISMISSED** [review]

**Review by ZKRobi — APPROVED** [review]

File: Tryouts/Prototypes/hackday1/ComposeUI.Example.DataService/Publisher.cs [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+                    }
+                    catch (Exception exception)
+                    {
+                        _logger?.LogError(exception.ToString());
+                    }
+                } 
+            }
+        }
+    }
+    internal class Publisher
```
[the remark this answers is not in the document]
Comment (kruplm) [reply Tryouts/Prototypes/hackday1/ComposeUI.Example.DataService/Publisher.cs:27]: Hi Marton! I followed Microsoft's suggestions on how to implement the Observer Design Pattern.
https://docs.microsoft.com/en-us/dotnet/standard/events/observer-design-pattern 
It's using System.IObserver.

That's why I have the SubscriptionsMonitor - which I know, could have a better name.

Next time I touch the code I'm going to look into System.Reactive - the approach looks similar to RxJS - and the other comments you had.

[the remark this answers is not in the document]
Comment (BalassaMarton) [reply Tryouts/Prototypes/hackday1/ComposeUI.Example.DataService/Publisher.cs:27]: System.Reactive IS Rx for 
.NET. Start here: https://github.com/dotnet/reactive

**Pull request merged**